File: n-repeated-element-in-size-2n-array.ts
{
    function repeatedNTimes(nums: number[]): number {
        const hash = new Map();
        const search = nums.length / 2

        return nums.find(num => {
            const count = (hash.get(num) || 0) + 1

            hash.set(num, count)
            
            if(count === search) {
                return num
            }
        }) || 0
    };

    console.log(repeatedNTimes([5,1,5,2,5,3,5,4]))
}

File: n-repeated-element-in-size-2n-array.rs
pub fn repeated_n_times(nums: Vec<i32>) -> i32 {
    for i in 1..nums.len() {
        if nums[i] == nums[i - 1] {
            return nums[i];
        }
    }

    if nums[1] == nums[3] {
        nums[1]
    } else {
        nums[0]
    }
}
